packages/good-design/src/apps/onramp/OnramperController.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Onramper } from "./Onramper";
+import { useEthers, useEtherBalance, useTokenBalance, useCall } from "@usedapp/core";
+import { WebViewMessageEvent } from "react-native-webview";
+import { useContractFunctionWithDefaultGasFees, useGetContract } from "@gooddollar/web3sdk-v2";
+import * as ethers from "ethers";
+import { Contract } from "ethers";
+import { useModal } from "../../hooks/useModal";
+import { View, Text } from "native-base";
+import { WalletAndChainGuard } from "../../core";
+import { useSignWalletModal } from "../../hooks/useSignWalletModal";
+
+export const OnramperController = () => {
+  const cusd = "0x765de816845861e75a25fca122bb6898b8b1282a";
```

Comment:
basically all the contract code and related logic that i've done should be moved to the sdk package.
meaning the sdk should export a hook that returns the swap/createAndSwap and a method to trigger the "fetch"

Reply:
Moved to sdk-v2, one hook useBuyGd and on utility function for fetching swapHelper response